### PR TITLE
Make backend authoring more friendly

### PR DESCRIPTION
**Pull request by pixelherodev**
Extracted pieces from SPU II PR which are useful for other backends

### CODE REVIEW

File: lib/std/target.zig [hunk 1]
```diff
@@ -1286,10 +1301,14 @@ pub const Target = struct {
                 .nvptx64,
                 => return result,
 
-                // TODO go over each item in this list and either move it to the above list, or
+                // These architectures have been confirmed as not supporting Linux.
+                .avr,
+                .spu_2,
+                => unreachable,
```
Comment (andrewrk) [inline lib/std/target.zig:1307]: These just go in the above list

File: src-self-hosted/link.zig [hunk 2]
```diff
@@ -35,21 +34,43 @@ pub fn openBinFilePath(
...
+            bin_file.* = try openBinFile(allocator, file, options);
+            return &bin_file.base;
+        },
+        //        .hex => {
+        //            var bin_file = try allocator.create(File.Hex);
+        //            errdefer allocator.destroy(bin_file);
+        //            bin_file.* = try openHexFile(allocator, file, options);
+        //            return &bin_file.base;
+        //        },
+        else => |e| {
```
Comment (andrewrk) [inline src-self-hosted/link.zig:70]: pls no commented out code

File: src-self-hosted/link.zig [hunk 3]
```diff
@@ -560,25 +607,26 @@ pub const File = struct {
 
         pub fn populateMissingMetadata(self: *Elf) !void {
             const small_ptr = switch (self.ptr_width) {
-                .p32 => true,
+                .p16, .p32 => true,
                 .p64 => false,
             };
```
Comment (andrewrk) [inline src-self-hosted/link.zig:612]: ELF doesn't have a 16 bit format. The changes here look kinda smelly. What do you think about having a separate struct for this ELF16 format (along with link.Elf and link.C)?

File: src-self-hosted/link.zig [hunk 4]
```diff
@@ -1436,15 +1486,22 @@ pub const File = struct {
             const endian = self.base.options.target.cpu.arch.endian();
             const off = shdr.sh_offset + @as(u64, entry_size) * index;
             switch (self.ptr_width) {
+                .p16,
+                // TODO wtf
+                //=>// {
+                //var buf: [2]u8 = undefined;
+                //mem.writeInt(u16, &buf, @intCast(u16, self.offset_table.items[index]), endian);
+                //try self.base.file.?.pwriteAll(&buf, off);
+                //},
```
Comment (andrewrk) [inline src-self-hosted/link.zig:1495]: looks like this is still WIP

**Comment by pixelherodev** [comment]
Closing in favor of #5970, #5971, and #5972.

**Pull request closed without merging**